src/webots/core/WbSysInfo.cpp

```diff
@@ -498,4 +498,35 @@ bool WbSysInfo::isAmdLowEndGpu(QOpenGLFunctions *gl) {
   return false;
 }
 
+#else
+
+bool WbSysInfo::isLowEndGpu() {
+  static char lowEndGpu = -1;  // not yet determined
+  if (lowEndGpu == -1) {       // based on the telemetry data from https://cyberbotics.com/telemetry
+    lowEndGpu = 0;
+    const QString &renderer = openGLRenderer();
+    if (renderer.contains("Intel") && renderer.contains(" HD Graphics ")) {
+      // we support only recent Intel GPUs from about 2015
+      if (renderer.contains("Ivybridge") || renderer.contains("Sandybridge") || renderer.contains("Haswell") ||
+          renderer.contains("Ironlake"))
+        lowEndGpu = 1;
+      else {
+        const int modelIndex = renderer.indexOf(" HD Graphics ") + 13;
```

Comment:
Can we use regex here? It would be more readable

Reply:
You are right, this is much more readable with regex.